### PR TITLE
Issue #35: satelite projects should use latest checkstyle as soon at is appears in maven repo 

### LINKED ISSUE #35: satelite projects should use latest checkstyle as soon at is appears in maven repo
Right now we do bump a version in a few projects https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release#update-for-released-version-reference right after release.

Let's find a way to define in pom.xml the latest snapshot and most recent version. And the latest released version when we define dependencies. To let us skip such updates at all.

CI of such projects should be able to get the latest version from https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml and run projects with such version.

We can have in all such repos, a GitHub action that runs once a day check https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml, put such values in required files, and send PRs with changes (PR is necessary as we need notice is any problems happens if we introduce breaking changes)

TASK:

- [ ] Version Bump in `ivy.xml`
- [ ] Version Bump in `pom.xml`
- [ ] Version Bump in `build.gradle`

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Partially Fixes #35

### CODE REVIEW

File: .github/workflows/bump_project_version.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#############################################################################
+# GitHub Action to Bump Project Version.
+#
+# Workflow starts every day at the end of the day.
+#
+#############################################################################
+name: "Bump Project Version"
```
Comment (romani) [inline .github/workflows/bump_project_version.yml:7]: Bump CS version in repo

File: .github/workflows/bump_project_version.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  schedule:
+    - cron: '0 0 * * *'
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  bump:
+    name: Bump Project Version
```
Comment (romani) [inline .github/workflows/bump_project_version.yml:16]: apply comments from https://github.com/checkstyle/contribution/pull/635#pullrequestreview-1079599409

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .github/workflows/bump_project_version.yml (hunk 1 above)
Comment (Rahulkhinchi03) [reply .github/workflows/bump_project_version.yml]: Changes Done

File: .github/workflows/bump_project_version.yml (hunk 2 above)
Comment (Rahulkhinchi03) [reply .github/workflows/bump_project_version.yml]: Changes Done

**Review by romani — APPROVED** [review]
ok lets try

**Pull request merged**